### DISCUSSION TITLE
ValKey as a replacement of Redis

### DISCUSSION THREAD

**Discussion by arpan57** [valkey-io/valkey#722] [Q&A] [answered]
Hello community, 

I am looking to replace Redis (clusters) with a potential alternative solution. I noticed that the ValKey is a fork of Redis 7.2.5. 

I wonder if anyone has successfully replaced Redis servers/clusters with ValKey. How about user's migration and developer's API compatibility? Do developers need to re-write the code or just replace the binary libraries and re-compile the code? 

Any other issues we should be aware for replacing Redis by ValKey? 
Any pointer or documentation of steps for Redis to ValKey migration. 

Apologies if too many requests in a single question. 

Many thanks, 
Arpan

**Comment by PingXie** [comment, marked as answer]
Hi @arpan57, Valkey 7.2 is fully compatible with Redis 7.2 and drop-in replacement is fully supported. There is no need to change any of your application code.

**Reply by TeddyAlbina** [reply]
Hello does Valkey have the same HA system with sentinels ?

**Reply by madolson** [reply]
Yes, as the Valkey code is forked off of Redis, it works the same way.
